documentation/general/workloads/cross-version-workloads.md

```diff
@@ -0,0 +1,138 @@
+
+```xml
+  <ImportGroup Condition="'$(RunAOTCompilation)' == 'true' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetFrameworkVersion)' == '6.0'">
+    <Import Project="Sdk.props" Sdk="Emscripten.Sdk.net6" />
+  </ImportGroup>
+  <ImportGroup Condition="'$(RunAOTCompilation)' == 'true' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetFrameworkVersion)' == '7.0'">
+    <Import Project="Sdk.props" Sdk="Emscripten.Sdk.net7" />
+  </ImportGroup>
+```
+
+## .NET 6 manifests in .NET 7 SDK
+
+The workload manifests are included in the .NET SDK.  This presents an issue for the runtime-specific workload manifests, as they would naturally be produced from the branch of the runtime they help target, but we don't want to introduce cross-branch build dependencies.
+
+Our solution to this will be to create copies of these runtime-specific manifests in branches corresponding to later SDK versions.  For example, the `workload.emscripten.net6` manifest would naturally be produced out of the .NET 6 branch.  It would use the 6.0.100 feature band, and would have a version number such as 6.0.0.  However, for consumption in the .NET 7 SDK, we would *also* produce a version of the `workload.emscripten.net6` manifest out of the .NET 7 branch.  In that case it would use the 7.0.100 feature band, and would have a version number such as 7.0.0.
+
+The workload packs referred to by this manifest would only be produced out of the .NET 6 branch.  The packs themselves are not used when building the .NET SDK.  They are needed in order to install the corresponding workload, so during testing of builds the right NuGet feeds will need to be set up in order to access the .NET 6 assets.  This is similar to what we already have today where the right feeds need to be set up in order to test self-contained builds.
+
+We will need to keep the versions of the runtime-specific workload packs synced between the .NET 6 and .NET 7 versions of the runtime-specific workload manifests.  During previews for .NET 7, this will be a manual process, where we will update the versions in the .NET 7 manifest manually when a new servicing version of .NET 6 is released.  After .NET 7 is released, we hope to be able to automatically calculate the .NET 6 version numbers by adding a fixed amount to the .NET 7 patch version.  For example, if .NET 7.0.1 releases together with .NET 6.0.12, then we can add 11 to the patch version of .NET 7 to calculate the corresponding version for the .NET 6 runtime-specific packs.  This would let us avoid having to manually sync the version numbers for each servicing release.
```

Comment:
> After .NET 7 is released, we hope to be able to automatically calculate the .NET 6 version numbers by adding a fixed amount to the .NET 7 patch version.

Once the initial release is done, would this number be frozen? I naively expect this to be the case, because from the time that 7.0.1 is released there would likely be runtime releases for both 6.0.x and 7.0.x from that time onward.

How does EOL/servicing lifetimes impact this? If the number 11 is hard-coded but 6.0.x runtimes are published longer than 7.0.x runtimes for example, would we be forced to come back and do more complex logic here?

Reply:
We chatted about this today and the takeaway was we're fine with this amount of book-keeping. We will default to an offset-based version calculation, but use that offset to compute a default value for the specific version offsets per-runtime. Then, if we need to pin specific offsets we can simply set the matching properties to specific values.